proxy/http2/Http2ClientSession.cc

```diff
@@ -326,6 +327,9 @@ Http2ClientSession::main_event_handler(int event, void *edata)
     break;
 
   case VC_EVENT_WRITE_COMPLETE:
+    if (this->get_half_close_flag()) {
```

Comment:
Yes, we should try that. The point here is how much the  "as long as possible" is.

\> 1.
We should stop accepting new steams. The reason is we could not increase the last stream id.

> Endpoints MUST NOT increase the value they send in the last stream identifier, since the peers might already have retried unprocessed requests on another connection

So we should ignore the frame. Probably it's better to send RST_STREAM frame.

\> 2.
We should also close the session by active timeout or inactive timeout. Even if there're outstanding streams, we should not leave the connection for long time after sending GOAWAY frame. Especially when TS is handling Connection Error.